lib/rspec/expectations/fail_with.rb

```diff
@@ -24,8 +24,7 @@ def fail_with(message, expected=nil, actual=nil)
                                "appropriate failure_message[_when_negated] method to return a string?"
         end
 
-        diff = differ.diff(actual, expected)
-        message = "#{message}\nDiff:#{diff}" unless diff.empty?
+        message = ::RSpec::Matchers::ExpectedsForMultipleDiffs.from(expected).message_with_diff(message, method(:differ), actual)
```

Comment:
I think I noticed some problems with that, it seemed like differ wasn't stateless, i.e. the more you use it, more weirder result you get. But could be I wrong, I will try and see if it possible.